solc/CommandLineInterface.cpp

```diff
@@ -752,11 +753,12 @@ map<string, Json::Value> CommandLineInterface::parseAstFromInput()
 			astAssert(ast["sources"][src][astKey]["nodeType"].asString() == "SourceUnit",  "Top-level node should be a 'SourceUnit'");
 			astAssert(sourceJsons.count(src) == 0, "All sources must have unique names");
 			sourceJsons.emplace(src, move(ast["sources"][src][astKey]));
-			tmpSources[src] = util::jsonCompactPrint(ast);
+			tmpSource[src] = util::jsonCompactPrint(ast); // could this potentially override pre-existing sources?
```

Comment:
Judging by the function name (`parseAstFromInput()`), I guess it could?

Interesting, I did not know we had a feature like this. Maybe I should mention it in #11105?